### PR TITLE
Allow specifying cmake setup directory

### PR DESCRIPTION
**Pull request by ryxli**
# Description

Please include a brief summary of the changes, relevant motivation and context.

Allow specifying which directory to store cmake build cache, which makes user defined docker buildx workflows more flexible.

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [x ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x ] The functionality is complete
- [na] I have commented my code, particularly in hard-to-understand areas
- [na ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [na] I have added tests that prove my fix is effective or that my feature works
- [na] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by ptrendx** [comment]
Hello @ryxli, could you please sign your commits (https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst#sign-your-work)?

File: build_tools/build_ext.py [hunk 1]
```diff
@@ -106,7 +106,9 @@ def run(self) -> None:
                 if isinstance(ext, CMakeExtension):
                     print(f"Building CMake extension {ext.name}")
                     # Set up incremental builds for CMake extensions
-                    setup_dir = Path(__file__).resolve().parent.parent
+                    setup_dir = Path(
+                        os.getenv("CMAKE_SETUP_DIR", Path(__file__).resolve().parent.parent)
+                    )
                     build_dir = setup_dir / "build" / "cmake"
```
Comment (timmoon10) [inline build_tools/build_ext.py:112]: Some minor tweaks:
- Prefixing envvar with `NVTE_` helps avoid conflicts with CMake envvars
- We should build directly in the build dir if provided, not in a `build/cmake` subdirectory
```suggestion
                    build_dir = os.getenv("NVTE_CMAKE_BUILD_DIR")
                    if build_dir:
                        build_dir = Path(build_dir).resolve()
                    else:
                        root_dir = Path(__file__).resolve().parent.parent
                        build_dir = root_dir / "build" / "cmake"
```

Comment (ryxli) [reply build_tools/build_ext.py]: good suggestions, thanks

**Review by timmoon10 — APPROVED** [review]

**Pull request merged**